src/app/store/accounts/blockchain/bitcoin/native-segwit-account.hooks.ts

```diff
@@ -147,10 +148,17 @@ export function useUpdateLedgerSpecificNativeSegwitUtxoHexForAdddressIndexZero()
         )
       )
     );
+
     inputSigningConfig.forEach(({ index }) => {
-      tx.updateInput(index, {
-        nonWitnessUtxo: Buffer.from(inputsTxHex[index], 'hex'),
-      });
+      // don't update nonWitnessUtxo if it already exists
+      if (tx.data.inputs.every(input => !isDefined(input.nonWitnessUtxo))) {
+        tx.updateInput(index, {
+          nonWitnessUtxo: Buffer.from(inputsTxHex[index], 'hex'),
+        });
```

Comment:
Still some confusion on this.

_What we want to happen_: if a given input, we've been asked to sign, already has a `nonWitnessUtxo`, then don't add a new one

_What happens here_: if all of the inputs in the transaction don't have `nonWitnessUtxo`, then don't add a new one

This means that, if some unrelated input in the transaction doesn't have a `nonWitnessUtxo`, then we don't add it to any. We need to inspect the state of _a given input_, not the whole transaction.

The earlier discussion spoke of `getInput` or `tx.inputs[index].

```ts
if (!tx.data.inputs[index].nonWitnessUtxo) {
```

Reply:
Thanks, that's what I was trying to achieve but let the 🐰 get in my head and I got confused

Reply:
I think I have it correctly now. I spent some time working on adding a unit test too but I will need more time to add mocks and do a refactor of `useUpdateLedgerSpecificNativeSegwitUtxoHexForAdddressIndexZero`. 

I can continue with that next week

Reply:
Also curious what part of `useUpdateLedgerSpecificNativeSegwitUtxoHexForAdddressIndexZero` needs refactoring?

Reply:
It doesn't need refactoring but I wanted to add a unit test to make sure its working properly and to do that I would need to refactor it to as its tricky to unit test hooks but I can try and add a test that covers the hook .

Reply:
I think it's solved now. Apologies for the back and forth. I thought we needed to have a nested loop as I didn't understand we could use the same `index` from `inputSigningConfig`